LASzip: fix dylib names for darwin and so on

natsukium opened the pull request:
## Description of changes

LASzip had a relative dylib name in darwin, so downstream packages like python3Packages.laszip could not find it.
This PR fixes that and also bumps the related package.

- python310Packages.laszip: 0.2.1 -> 0.2.3
- python310Packages.laspy: 2.4.1 -> 2.5.1

I found this build failure in #246143

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

natsukium committed:
LASzip: fix dylib name for darwin

natsukium committed:
LASzip: refactor

- use finalAttrs
- replace hash with SRI hash
- add meta.changelog

natsukium committed:
python310Packages.laszip: refactor

- remove unused inputs
- move pybind11 to nativeBuildInputs
- set dontUseCmakeConfigure
- add meta.changelog

natsukium committed:
python310Packages.laspy: refactor

- add setuptools to nativeBuildInputs
- replace checkInputs with nativeCheckInputs
- fix the changelog url

natsukium committed:
python310Packages.laspy: 2.4.1 -> 2.5.1

Changelog: https://github.com/laspy/laspy/blob/2.5.1/CHANGELOG.md

natsukium commented:
Result of `nixpkgs-review` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>LASzip</li>
    <li>grass</li>
    <li>pdal</li>
    <li>python310Packages.laspy</li>
    <li>python310Packages.laspy.dist</li>
    <li>python310Packages.laszip</li>
    <li>python310Packages.laszip.dist</li>
    <li>python311Packages.laspy</li>
    <li>python311Packages.laspy.dist</li>
    <li>python311Packages.laszip</li>
    <li>python311Packages.laszip.dist</li>
  </ul>
</details>

risicle commented:
laszip fails to build on linux.

```
/build/source/src/lasunzipper.cpp:79:46: error: ‘numeric_limits’ is not a member of ‘std’
     79 |     size_t max_bytes_in_output_stream = std::numeric_limits<std::stringstream::int_type>::max();
```

Missing an include? Not choosing a modern enough c++ std?

natsukium commented:
Thanks for checking!

> Missing an include? Not choosing a modern enough c++ std?

`#include <limits>` seems to be missing, but I didn't get this error in Ubuntu.
I will check the c++ standard and if it does not solve this, I will apply the patch.

natsukium committed:
python310Packages.laszip: 0.2.1 -> 0.2.3

Diff: tmontaigu/laszip-python@0.2.1...0.2.3

Changelog: https://github.com/tmontaigu/laszip-python/blob/0.2.3/Changelog.md

natsukium commented:
The build on linux seems to be fine now.

risicle commented on pkgs/development/python-modules/laszip/default.nix:
```diff
@@ -1,48 +1,44 @@
+    hash = "sha256-MiPzL9TDCf1xnCv7apwdfcpkFnBRi4PO/atTQxqL8cw=";
   };
 
+  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.isLinux "-std=c++17";
```
This should probably be `stdenv.cc.isGNU` rather than `stdenv.isLinux` - it's about the compiler being gcc rather than what kernel it's running on.

natsukium replied:
Good catch.
That's right, Thanks!

risicle reviewed (approved):
LGTM, `nixpkgs-review` happy, macos 10.15 & nixos x86_64
